src/core/backbone_router/local.cpp

```diff
@@ -275,49 +271,88 @@ otError Local::RemoveDomainPrefix(const otIp6Prefix &aPrefix)
                                            BitVectorBytes(aPrefix.mLength)) >= aPrefix.mLength,
                  error = OT_ERROR_NOT_FOUND);
 
-    mDomainPrefixConfig.mPrefix.mLength = 0;
-
     if (IsEnabled())
     {
-        SuccessOrExit(error =
-                          Get<NetworkData::Local>().RemoveOnMeshPrefix(aPrefix.mPrefix.mFields.m8, aPrefix.mLength));
+        error = Get<NetworkData::Local>().RemoveOnMeshPrefix(aPrefix.mPrefix.mFields.m8, aPrefix.mLength);
+        LogDomainPrefix("Remove", error);
     }
 
+    mDomainPrefixConfig.mPrefix.mLength = 0;
```

Comment:
It seems possible that this function can return some error but still have `mDomainPrefixConfig.mPrefix.mLength = 0;
` executed. Normally we expect nothing has changed when a function returns error; Should we just set `error = OT_ERROR_NONE` here? Or should we change this function to return `void` just like `Local::SetDomainPrefix` ?

Reply:
yes, should recover error state. 
Different from SetDomainPrefix() before which the prefix has been pre-checked. 
Prefer to return error here.

Reply:
> Different from SetDomainPrefix() before which the prefix has been pre-checked.

No string opinion on return values. But is it better for callee not to assume how it is called by callers?